docs/developer/release-team.md

```diff
@@ -0,0 +1,70 @@
+
+As noted above, making changes to  the CAPI release cadence is out of scope for this initial release team process. 
+
+## Specific Responsibilities
+
+- Release patch releases monthly or more often as needed, so users will get fixes & updated dependencies with CVE fix on a predictable cadence
+- Release quarterly (to be revised, see future work) minor releases, so users will get new features on a predictable cadence
+- Monitor CI signal, so a release can be cut at any time, and add CI signal for each new release branch
+- Maintain and improve user facing documentation about releases, release policy and calendar
+- Update the CAPI Netlify book certificates and DNS
+- Update the clusterctl Homebrew formula
+- Create and maintain the GitHub release milestone
+- Maintain and improve and release automation, tooling & related developer docs
+- Track tasks needed to add support for new Kubernetes versions
+
+
+## Team Roles
+
+- **Release Lead**: responsible for coordinating release activities, assembling the release team, taking ultimate accountability for all release tasks to be completed on time, and ensuring that a retrospective happens. The lead is also responsible for ensuring a successor is selected and trained for future release cycles.
+- **Communications/Docs/Release Notes Manager**: Responsible for communicating key dates to the community, improving release process documentation, and polishing release notes. Also responsible for ensuring the user-facing Netlify book and provider upgrade documentation are up to date.
+- **CI Signal/Bug Triage/Automation Manager**: Assumes the responsability of the quality gate for the release and makes sures blocking issues and bugs are triaged and dealt with in a timely fashion. Helps improve release automation and tools.
+- **Shadow**: Any Release Team member may select one or more mentees to shadow the release process in order to help fulfill future Release Team staffing requirements and continue to grow the CAPI community in general.
+
+## Team Selection
+
+To start, the release team will be assembled by the release team lead based on volunteers. A call for volunteers can be made through the usual communication channels (office hours, Slack, mailing list, etc.). In the future, we may consider introducing an application process similar to the Kubernetes release team application process. 
+
+### Selection Criteria
+
+When assembling a release team, the release team lead should look for volunteers who:
+
+- Can commit to the amount of time required across the release cycle
+- Are enthusiastic about being on the release team
+- Are members of the Cluster Lifecycle SIG
+- Have some prior experience with contributing to CAPI releases (such as shadowing a role for a prior release)
+- Have diverse company affiliations (i.e. not all from the same company)
+
+## Time Commitment
```

Comment:
Might be in a follow-up iteration, but I would like to see two sections in the paragraph:
- one for individuals willing to contribute, explaining why this could be an interesting and hopefully fun experience
- one for helping people willing to contribute in making the case with their managers, explaining why this is important for companies leveraging in CAPI

Reply:
Added a paragraph below